selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/RecreateUpdate.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.hotupdate;
+
+import static java.lang.Integer.*;
+import static org.testng.Assert.*;
+
+import com.google.inject.Inject;
+import java.io.IOException;
+import org.eclipse.che.commons.json.JsonHelper;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.TestGroup;
+import org.eclipse.che.selenium.core.executor.OpenShiftCliCommandExecutor;
+import org.eclipse.che.selenium.core.provider.CheTestApiEndpointUrlProvider;
+import org.eclipse.che.selenium.core.requestfactory.CheTestAdminHttpJsonRequestFactory;
+import org.eclipse.che.selenium.core.utils.WaitUtils;
+import org.eclipse.che.selenium.core.utils.process.ProcessAgent;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CheTerminal;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@Test(groups = {TestGroup.OSIO, TestGroup.MULTIUSER})
+public class RecreateUpdate {
+  private static final String COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE =
+      "get dc | grep che | awk '{print $2}'";
+
+  @Inject CheTestAdminHttpJsonRequestFactory testUserHttpJsonRequestFactory;
+  @Inject CheTestApiEndpointUrlProvider cheTestApiEndpointUrlProvider;
+  @Inject ProjectExplorer projectExplorer;
+  @Inject OpenShiftCliCommandExecutor openShiftCliCommandExecutor;
+  @Inject private ProcessAgent processAgent;
+  @Inject private TestWorkspace workspace;
+  @Inject private Ide ide;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private CheTerminal terminal;
+  @Inject private Menu menu;
+
+  private int cheDeploymentBeforeRollout;
+
+  private enum WsMasterStauses {
```

Comment:
It looks like useful common enum and maybe it should be stored separately or in related page object.

Reply:
This enum describes statuses of ws-master on OpenShift . On this moment this one is used in this place. But is we will need this one for other tests, of course we should store this into separate class.